server/src/main/java/org/opensearch/index/compositeindex/datacube/DimensionFactory.java

```diff
@@ -36,14 +36,15 @@ public class DimensionFactory {
     public static Dimension parseAndCreateDimension(
         String name,
         String type,
+        Boolean isUnsignedLong,
```

Comment:
This is not extensible, lets go with a separate dimension for unsigned long?

That should solve `isUnsignedLong` problem easily

Reply:
Done. We now have a new Dimension - `UnsignedLongDimension` which extends `Numeric Dimension`.